nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java

```diff
@@ -1004,7 +1004,7 @@ public ProcessorNode createProcessor(final String type, String id, final boolean
             creationSuccessful = false;
         }
 
-        final ValidationContextFactory validationContextFactory = new StandardValidationContextFactory(controllerServiceProvider);
+        final ValidationContextFactory validationContextFactory = new StandardValidationContextFactory(controllerServiceProvider, this);
```

Comment:
I really don't like passing the FlowController itself. Why can't it just pass the value? Or if it needs to be able to be updated, it could pass a shared reference.